File: playground/examples/python/routing.py
from sayiir import task, Flow, run_workflow

@task
def classify(ticket: dict) -> str:
    return "billing" if ticket["type"] == "invoice" else "tech"

@task
def handle_billing(ticket: dict) -> str:
    return f"Billing handled: ticket #{ticket['id']}"

@task
def handle_tech(ticket: dict) -> str:
    return f"Tech resolved: ticket #{ticket['id']}"

workflow = (
    Flow("support")
    .route(classify, keys=["billing", "tech"])
        .branch("billing", handle_billing)
        .branch("tech", handle_tech)
    .done()
    .build()
)
print(run_workflow(workflow, {"id": 42, "type": "invoice"}))

File: playground/examples/node/routing.js
const { task, flow, runDurableWorkflow, InMemoryBackend } = require("sayiir");

const classify = task("classify", (ticket) => {
  return ticket.type === "invoice" ? "billing" : "tech";
});

const handleBilling = task("handle-billing", (ticket) => {
  return `Billing handled: ticket #${ticket.id}`;
});

const handleTech = task("handle-tech", (ticket) => {
  return `Tech resolved: ticket #${ticket.id}`;
});

const workflow = flow("support")
  .route(classify, ["billing", "tech"])
    .branch("billing", handleBilling)
    .branch("tech", handleTech)
  .done()
  .build();

const backend = new InMemoryBackend();
const status = runDurableWorkflow(workflow, "run-1", { id: 42, type: "invoice" }, backend);
console.log(status.output);
